### PR TITLE
Elgino

### PR DESCRIPTION
**Pull request by tmagicturtle**

### CODE REVIEW

File: data/mods/ssb/abilities.js [hunk 1]
```diff
@@ -223,6 +223,19 @@ let BattleAbilities = {
...
+		shortDesc: "Defense +1 and heal 20% after hit by contact move.",
+		onAfterDamage(damage, target, source, effect) {
+			if(effect && effect.flags['contact']) {
+				this.boost({def: 1});
+				this.heal(target.maxhp / 5);
+			}
+		},
+		id: "giblovepls",
+		name: "Gib love pls",
+	}
```
Comment (ghost) [inline data/mods/ssb/abilities.js]:
```suggestion
	},
```

File: data/mods/ssb/random-teams.js [hunk 2]
```diff
@@ -215,6 +215,12 @@ class RandomStaffBrosTeams extends RandomTeams {
 				signatureMove: 'Ancestral Power',
 				evs: {hp: 252, spd: 4, spe: 252}, nature: 'Timid',
 			},
+			'Elgino': {
+				species: 'Mimikyu', ability: 'Gib love pls', item: ['Mimikium Z', 'Ghostium Z', 'Fightinium Z'], gender: '',
+				moves: ['Spectral Thief', 'Play Rough', ['Shadow Sneak', 'Swords Dance']],
+				signatureMove: 'Rough Snuggle',
+				evs: {hp: 4, atk: 252, spe: 252}, nature: 'Adamant',
```
Comment (HoeenCoder) [inline data/mods/ssb/random-teams.js]: needs to be shiny

**Review by HoeenCoder — COMMENTED** [review]
Missing animation on custom move.

File: data/mods/ssb/abilities.js [hunk 3]
```diff
@@ -223,6 +223,19 @@ let BattleAbilities = {
...
 		},
 	},
+	// Elgino
+	giblovepls: {
+		desc: "This Pokemon's Defense is raised 1 stage and heals 20% after it is damaged by a contact move.",
+		shortDesc: "Defense +1 and heal 20% after hit by contact move.",
+		onAfterDamage(damage, target, source, effect) {
+			if (effect && effect.flags['contact']) {
+				this.boost({def: 1});
+				this.heal(target.maxhp / 5);
```
Comment (HoeenCoder) [inline data/mods/ssb/abilities.js]: Specify the pokemon to heal/boost

File: data/mods/ssb/moves.js [hunk 4]
```diff
@@ -1018,6 +1018,32 @@ let BattleMovedex = {
...
+		desc: "Power doubles if the target is not fully evolved.",
+		shortDesc: "Power doubles if the target is NFE.",
+		id: "roughsnuggle",
+		name: "Rough Snuggle",
+		isNonstandard: "Custom",
+		pp: 10,
+		priority: 0,
+		flags: {protect: 1, mirror: 1},
+		onPrepareHit(target, source) {
+			this.add('-anim', source, 'Let\'s Snuggle Forever', target);
```
Comment (HoeenCoder) [inline data/mods/ssb/moves.js:1038]: Just like with all other custom moves, you need to cancel the default animation first.
```javascript
onTryMove() {
	this.attrLastMove('[still]');
},
```

**Review by HoeenCoder — COMMENTED** [review]
Looks good

File: data/mods/ssb/statuses.js [hunk 5]
```diff
@@ -397,6 +397,18 @@ let BattleStatuses = {
 			this.add(`c|@Eien|This game is Bad Civilization...`);
 		},
 	},
+	elgino10: {
```
Comment (ghost) [inline data/mods/ssb/statuses.js]: er....

File: data/mods/ssb/abilities.js (hunk 3 above)
Comment (ghost) [reply data/mods/ssb/abilities.js]: What he means basically is just to add `, target` inside both. Or something like that

Comment (ghost) [reply data/mods/ssb/abilities.js]: Although, kung fu panda uses similar code and doesnt specify so that irks me a bit :V

Comment (HoeenCoder) [reply data/mods/ssb/abilities.js]: Then that needs to be updated too.

**Pull request merged**